src/webui/api/synccontroller.cpp

```diff
@@ -369,13 +381,33 @@ namespace
 //  - "up_rate_limit: upload speed limit
 //  - "queueing": priority system usage flag
 //  - "refresh_interval": torrents table refresh interval
+//  - "free_space_on_disk": Free space on the default save path
 // GET param:
 //   - rid (int): last response id
 void SyncController::maindataAction()
 {
     auto lastResponse = sessionManager()->session()->getData(QLatin1String("syncMainDataLastResponse")).toMap();
     auto lastAcceptedResponse = sessionManager()->session()->getData(QLatin1String("syncMainDataLastAcceptedResponse")).toMap();
 
+    if (!m_freeDiskSpaceThread) {
```

Comment:
I don't love that this block of code exists in an api controller function. I'm almost considering switching back to a timer and moving all this to the `SyncController` constructor, so that this function only ever has to read the value in `m_freeDiskSpace`. @glassez thoughts?